airbyte-webapp/src/core/form/schemaToFormBlock.ts

```diff
@@ -37,20 +38,49 @@ export const jsonSchemaToUiWidget = (
   }
 
   if (jsonSchema.oneOf?.length && jsonSchema.oneOf.length > 0) {
-    const conditions = Object.fromEntries(
-      jsonSchema.oneOf.map((condition) => {
-        if (typeof condition === "boolean") {
-          return [];
-        }
-        return [condition.title, jsonSchemaToUiWidget({ ...condition, type: jsonSchema.type }, key, path)];
-      })
+    let possibleConditionSelectionKeys = null as Set<string> | null;
```

Comment:
It would be, but if you do that, Typescript will auto-narrow down your type to just `null`:
<img width="918" alt="Screenshot 2022-12-15 at 11 54 04" src="https://user-images.githubusercontent.com/1508364/207841425-a3277063-390d-446b-92b8-1352a621c69b.png">